src/components/EventCreator/EventForm.tsx

```diff
@@ -0,0 +1,371 @@
+import React, { useState } from "react";
+import {
+  Button,
+  Container,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+  TextField,
+  Typography,
+  Stack,
+  FormControl,
+  InputLabel,
+  Box,
+  Card,
+  Switch,
+  FormControlLabel,
+} from "@mui/material";
+import Grid from '@mui/material/Grid2';
+import RadioButtonCheckedIcon from "@mui/icons-material/RadioButtonChecked";
+import CheckBoxIcon from "@mui/icons-material/CheckBox";
+import FormatListNumberedIcon from "@mui/icons-material/FormatListNumbered";
+import OptionsCard from "./OptionsCard";
+import { Option } from "../../interfaces";
+import { defaultRelays, signEvent } from "../../nostr";
+import { useNavigate } from "react-router-dom";
+import { useAppContext } from "../../hooks/useAppContext";
+import { useUserContext } from "../../hooks/useUserContext";
+import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
+import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
+import { DateTimePicker } from "@mui/x-date-pickers";
+import dayjs from "dayjs";
+import { useSigner } from "../../contexts/signer-context";
+import { useNotification } from "../../contexts/notification-context";
+import { NOTIFICATION_MESSAGES } from "../../constants/notifications";
+import { NOSTR_EVENT_KINDS } from "../../constants/nostr";
+
+export type PollTypes =
+  | "singlechoice"
+  | "multiplechoice"
+  | "rankedchoice"
+  | undefined;
+
+function generateOptionId() {
+  return Math.random().toString(36).substr(2, 9);
+}
+
+const pollOptions = [
+  {
+    value: "singlechoice",
+    icon: <RadioButtonCheckedIcon fontSize="small" />,
+    label: "Single Choice Poll",
+  },
+  {
+    value: "multiplechoice",
+    icon: <CheckBoxIcon fontSize="small" />,
+    label: "Multiple Choice Poll",
+  },
+  {
+    value: "rankedchoice",
+    icon: <FormatListNumberedIcon fontSize="small" />,
+    label: "Ranked Choice Poll",
+    disabled: true,
+  },
+];
+
+const EventForm = () => {
```

Comment:
Too much going on here, this component could be split into two child component, while the parent component "EventForm" can take the responsibility of coordinating between the two